tests/ui/impl-trait/issue-108592.rs

```diff
@@ -11,13 +11,10 @@ fn test_closure() {
     closure(&opaque());
 }
 
-mod helper {
-    pub type Opaque2 = impl Sized;
-    pub type Opaque<'a> = Opaque2;
-    fn define<'a>() -> Opaque<'a> {}
-}
-
-use helper::*;
+pub type Opaque2 = impl Sized;
+pub type Opaque<'a> = Opaque2;
+#[define_opaques(Opaque)]
+fn define<'a>() -> Opaque<'a> {}
```

Comment:
interesting test :thinking: can you add a note somewhere that we should look at the behavior of recursing into nested opaques this way somewhere? maybe just the tracking issue

Reply:
done, but the collector already had a FIXME, too